docs/error-codes.md

```diff
@@ -5,6 +5,12 @@ identification. The known codes are in the range 1000 to 6000. Custom
 steps should avoid using this range not to collide with existing or future
 error and warning codes.
 
+For versioned warnings, the warning version is indicated in parentheses following
+the error code. For example:
+
+#### `ILXXXX` (version): Message
```

Comment:
What is the added value in duplicating the version in every message? We could just group the warnings per warning with one line